lms/djangoapps/grades/models.py

```diff
@@ -361,30 +362,43 @@ def update_or_create_grade(cls, **params):
         """
         cls._prepare_params_and_visible_blocks(params)
 
+        first_attempted = params.pop('first_attempted')
         user_id = params.pop('user_id')
         usage_key = params.pop('usage_key')
-        attempted = params.pop('attempted')
 
         grade, _ = cls.objects.update_or_create(
             user_id=user_id,
             course_id=usage_key.course_key,
             usage_key=usage_key,
             defaults=params,
         )
-
-        if attempted and not grade.first_attempted:
-            grade.first_attempted = now()
+        if first_attempted is not None and grade.first_attempted is None:
+            if waffle.switch_is_active('grades_estimate_first_attempted'):
```

Comment:
Please use waffle_utils.py and its helper functions instead.  It provides essential namespacing and caching functionality.